include/image-commands.mk

```diff
@@ -223,6 +223,7 @@ define Build/check-size
 	[ $$limitsize -ge $$imagesize ] || { \
 		$(call ERROR_MESSAGE,    WARNING: Image file $@ is too big: $$imagesize > $$limitsize); \
 		rm -f $@; \
+		false; \
```

Comment:
I think the build in the build bots is intentionally not failing when the size check fails, it just does not create the image.

Reply:
I think so as well, as otherwise the whole target building would fail in buildbots and none of the other devices will get updated images.

Reply:
> I think the build in the build bots is intentionally not failing when the size check fails, it just does not create the image.

The problem I've seen is that broken images **_are still created_** even if the size check fails, because later steps in the device build "pipeline" do so (e.g. footer creation). Since no one is checking for the related error messages, this would go unnoticed and allow publishing broken images.

Allowing builds to continue on size-check failures isn't really a solution; those failures are meant to be acted on. Failure of the whole target build might seem like a problem, but it means the device issue is noticed and fixed (marked broken, image updated).

Perhaps it's possible to "abort" just the device build on error so no image is created, while allowing the rest of the target build to continue. Is that possible? In any case, we need to handle these errors better...

Reply:
instead of a `false` can it just be a `rm` so the bad image gets deleted but exit code remains good?

Reply:
well wait a sec, the `rm` is right there so why isn't it getting removed?

Reply:
this should be removing every possible thing that future build recipe steps rely on, maybe deleting the per device dtb as well will do it?

if that's too difficult, maybe create a stampfile that will be checked later as an independent make target in the rules for `make world`, in other words, create an actual make target called check-size, that gets called right before or after package/index is ran...

Reply:
also a possibility, if we create a stampfile it can be checked for directly after the build recipe

we have some parts of image.mk where

`Build/check-size`
is called right after
`concat_cmd`
so we probably just need more of that...